src/util/io.c

```diff
@@ -140,33 +143,55 @@ socket_close (
     return TSS2_RC_SUCCESS;
 }
 
+static void uinttostring(unsigned int num, char *str) {
```

Comment:
use itoa() or snprintf(), but don't reinvent the wheel please...

Reply:
OK, although generally libraries should not report errors, including using fprintf(). That's up to the application. But I see the rule is broken already for tpm2-tss, so I replaced it with snprintf().

Also itoa() does not exist in libc (although atoi does). Function itoa may exist in other libraries, but it is nonstandard.